src/components/SearchBar.ts

```diff
@@ -0,0 +1,127 @@
+import { Component, createElement } from "react";
+import { HybridConstraint, ListView, WrapperProps } from "./OfflineSearch";
+
+export interface SearchBarProps extends WrapperProps {
+    listView?: ListView;
+}
+
+interface SearchBarState {
+    buttonVisibility?: string;
```

Comment:
this is not a good state name `buttonVisibility` should be some name indicatie the there is a query.
The rendering will show button when there is a query.